components/intl/DateFromNow.tsx

```diff
@@ -20,12 +20,31 @@
 import * as React from 'react';
 import { DateSource, FormattedRelative } from 'react-intl';
 import { parseDate } from '../../helpers/dates';
+import { translate } from '../../helpers/l10n';
+import DateTimeFormatter from './DateTimeFormatter';
 
-interface Props {
+export interface DateFromNowProps {
   children?: (formattedDate: string) => React.ReactNode;
-  date: DateSource;
+  date?: DateSource;
 }
 
-export default function DateFromNow({ children, date }: Props) {
-  return <FormattedRelative value={parseDate(date)}>{children}</FormattedRelative>;
+export default function DateFromNow(props: DateFromNowProps) {
+  const { children = (f: string) => f, date } = props;
+
+  if (!date) {
+    // eslint-disable-next-line react/jsx-no-useless-fragment
+    return <>{children(translate('never'))}</>;
+  }
```

Comment:
```
if (!date) {
    return children(translate('never'));
  }
```
?

Reply:
Sadely no. That would make the component to possibly return something else than an `JSX.Element`, and causes issues where the component is consumed.